W3ChampionsStatisticService/Hubs/WebsiteBackendHub.cs

```diff
@@ -276,6 +276,33 @@ await Clients.Caller.SendAsync(
         }
     }
 
+    public async Task BlockPlayer(string battleTag)
+    {
+        var currentUser = _connections.GetUser(Context.ConnectionId)?.BattleTag;
+        if (currentUser == null)
+        {
+            return;
+        }
+        try
+        {
+            var friendList = await _friendCommandHandler.LoadFriendList(currentUser);
```

Comment:
Could the CanBlock be a function of the friend thingy so we don't have to load the entire friendlist?

Basically delegating the responsibility of fetching the necessary data to the function that evaluates the block, rather than the caller?

Reply:
Sure, but seems like a refactor out of scope of this